plugins/feeds/public/dfiq.py

```diff
@@ -91,6 +50,31 @@ class DFIQFeed(task.FeedTask):
         "description": "DFIQ feed",
     }
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._dfiq_kb = {}
+
+    def read_from_data_directory(self, directory: str) -> None:
```

Comment:
Don't you think we should make this method available at Task level to provide a readable stream to calling module? I think that could be useful for other module. We could also abstract this method with directory being a cloud storage for example by relying on [pyfilesystem](https://www.pyfilesystem.org/page/index-of-filesystems/)

Reply:
Yeah, we could! Although in this case, the function is very specific to how DFIQ is shaped (there's a directory with subdirectories, and then we parse YAML ignoring specific filetypes).

Having a "walk files from directory" abstracted to local FS, cloud, and others, would be great, but seem better suited for a dedicated PR.